internal/liveness/metrics_util.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2020 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package liveness
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ceph/ceph-csi/internal/util"
+
+	"k8s.io/klog"
+)
+
+const (
+	// csiConfigFile is the location of the CSI config file
+	csiConfigFile = "/etc/ceph-csi-config/config.json"
+)
+
+// CheckSubVolumeInCluster check if subvolume in cluster
+func CheckSubVolumeInCluster(monitors, subvolume, volumegroup, volume string, cr *util.Credentials) bool {
+	klog.V(4).Infof("cephfs: check exist of subvolume %s", subvolume)
+	stdout, _, err := util.ExecCommand(
+		"ceph",
+		"-m", monitors,
+		"--id", cr.ID,
+		"--keyfile="+cr.KeyFile,
+		"-c", util.CephConfigPath,
+		"fs", "subvolume", "ls", volume, volumegroup)
+	if err != nil {
+		klog.Errorf(fmt.Sprintf("failed to get subvolume list, command output: %s", err.Error()))
+		return false
+	}
+	if strings.Contains(string(stdout), subvolume) {
```

Comment:
this may fail if two subvolumes have the same prefix, better to split and do comparison

Reply:
I use func 'getVolumeRootPathCeph' to check subvolume